src/App.jsx

```diff
@@ -1,41 +1,76 @@
-import React from 'react';
-import './App.scss';
-import './styles/general.scss';
+import React, { useState, useEffect } from 'react';
+
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
+import { getUserPosts } from './api/posts';
+
+import './App.scss';
+import './styles/general.scss';
+
+const App = () => {
+  const [selectedUser, selectUser] = useState(0);
+  const [selectedPostId, setPostId] = useState(0);
+  const [posts, setPosts] = useState([]);
+
+  useEffect(() => {
+    const loadPosts = async() => {
+      const postsFromServer = await getUserPosts(selectedUser);
+
+      setPosts(postsFromServer);
+    };
```

Comment:
that kind of function could be created somewhere else and imported, no need to create it every time when your useEffect callback called